doc/samples/sample_notify_fifo.sh

```diff
@@ -60,6 +76,18 @@ if [[ ! -p $FIFO ]]; then
 	fi
 fi
 
+exec {FD}>>"$PID_FILE"
+if ! flock -e $FD --timeout 10; then
+	if ls -l /proc/$(cat "$PID_FILE")/fd | grep -qw "$(readlink -f "$PID_FILE")"
+	then
+		kill $(cat "$PID_FILE")
+	else
+		exit 1
+	fi
+	flock -e $FD --timeout 10 || exit 1
+fi
```

Comment:
Change the `if ... fi` block to:
```
if ! flock -e -n $FD; then
       OLD_PID=$(cat $PID_FILE)
       if ls -l /proc/$OLD_PID/fd | grep -qw "$(readlink -f "$PID_FILE")"
       then
               kill -QUIT $OLD_PID
       fi
       flock -e --timeout 10 $FD || exit 1
fi
```
`flock -e $FD --timeout 10` fails for me due to the options not being before the FD. Also we don't need to delay if the first flock cannot obtain the lock.

Saving the PID in $OLD_PID saves reading the PID file twice.

Sending SIGQUIT allows the old instance of the script to know why it is being signalled, and the EXIT trap to be executed.

The old instance of the script may be terminating while this part of the new instance is being executed, and so the old instance may remove the PID_FILE. This would mean that `ls -l /proc/$OLD_PID/fd` would fail, and we don't then want this instance of the script to terminate, so the `else exit 1` needs to be removed.

Also, move this block obtaining the lock to before checking for the existence of the FIFO to avoid a potential race of the old script removing the FIFO and this script creating the FIFO.